docs/PROVISIONING_ARCH.md

```diff
@@ -0,0 +1,77 @@
+# Automated Provisioning Architecture
+
+The CASL automated provisioning framework provides an opinionated architecture for handling automated provisioning of OpenShift clusters on any _automatable_ infrastructure. This document describes the architecture and principles
+
+## Four Phased Approach
+
+A provisioning run should be split into four (4) phases as represented in our [end-to-end.yml](/playbooks/openshift/end-to-end.yml) playbook:
+
+
+```yaml
+---
+# Provision Openstack Instances
+- import_playbook: provision-instances.yml <1>
+
+# Pre-Install Steps
+- import_playbook: pre-install.yml <2>
+
+# Install - this runs openshift-ansible
+- import_playbook: install.yml <3>
+
+# Post Install
+- import_playbook: post-install.yml <4>
+```
+
+1. _Provision Infrastructure_ creates the servers/instances and any other supporting infrastructure requires (network, DNS, load balancers, etc.). In general this phase
+  - Begins with having credentials with proper access to an Infrastructure provider.
+  - Ends with servers provisioned and running with an OS installed, SSH keys (with root or sudo) pushed, and `subscription-manager` installed.
```

Comment:
the `subscription-manager` installed step I believe is a Google Cloud only, and not sure I'd call it out here.